Demos/include/ApplicationBase.hpp

```diff
@@ -0,0 +1,598 @@
+
+        if (!excludedCommandLineArgs.count(DefaultArg::Uri))
+        {
+            _commandLineParser->Add<CommandlineParser::Option>(
+                defaultArgName.at(DefaultArg::Uri), "u", _arguments.registryUri, "[--registry-uri <silkitUri>]",
+                "-u, --registry-uri <silkitUri>: The registry URI to connect to. Defaults to '" + _arguments.registryUri
+                    + "'.");
+        }
+
+        if (!excludedCommandLineArgs.count(DefaultArg::Log))
+        {
+            _commandLineParser->Add<CommandlineParser::Option>(
+                defaultArgName.at(DefaultArg::Log), "l", "info", "[--log <level>]",
+                "-l, --log <level>: Log to stdout with level 'trace', 'debug', 'warn', 'info', 'error', 'critical' or "
+                "'off'. "
+                "Defaults to 'info' if the '--configuration' option is not specified. "
+                "Set to 'off' to explicitly turn off stdout logging. "
+                "Cannot be used together with '--configuration'.");
+        }
+
+        if (!excludedCommandLineArgs.count(DefaultArg::Config))
+        {
+            _commandLineParser->Add<CommandlineParser::Option>(
+                defaultArgName.at(DefaultArg::Config), "c", "", "[--config <filePath>]",
+                "-c, --config <filePath>: Path to the Participant configuration YAML or JSON file. "
+                "Cannot be used together with '--log'.");
+        }
+
+        if (!excludedCommandLineArgs.count(DefaultArg::Async))
+        {
+            _commandLineParser->Add<CommandlineParser::Flag>(defaultArgName.at(DefaultArg::Async), "a", "[--async]",
+                                                             "-a, --async: run in asynchronous mode. "
+                                                             "Cannot be used together with '--duration'.");
+        }
+
+        if (!excludedCommandLineArgs.count(DefaultArg::Duration))
+        {
+            _commandLineParser->Add<CommandlineParser::Option>(
+                defaultArgName.at(DefaultArg::Duration), "d", "1000", "[--duration <sim step duration in us>]",
+                "-d, --duration <sim step duration in us>: The step size in microseconds of the participant. "
+                "Defaults to 1000us. "
+                "Cannot be used together with '--async'.");
+        }
+
+        if (!excludedCommandLineArgs.count(DefaultArg::AsFastAsPossible))
+        {
+            _commandLineParser->Add<CommandlineParser::Flag>(
+                defaultArgName.at(DefaultArg::AsFastAsPossible), "f", "[--fast]",
+                "-f, --fast: Run the simulation as fast as possible. By default, the execution is slowed down "
+                "to a single work cycle per second."
+                "Cannot be used together with '--config'.");
+        }
+    }
+
+    auto ToLowerCase(std::string s) -> std::string
+    {
+        std::transform(s.begin(), s.end(), s.begin(), [](unsigned char c) { return (unsigned char)std::tolower(c); });
```

Comment:
The `std::string` type is a container of `char`, not `unsinged char`. You can also use `std::for_each` which iterates over all range elements.

```suggestion
        std::for_each(s.begin(), s.end(), [](char & c) { c = std::tolower(c); });
```